aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/UpdateHandler.java

```diff
@@ -112,14 +117,24 @@ public UpdateHandler(TagHelper tagHelper) {
             logger.log(String.format("INFO Updated tags for patch baseline %s %n", baselineId));
 
             // Set to default patch baseline
-            if (BooleanUtils.isTrue(model.getDefaultBaseline()) && !BooleanUtils.isTrue(previousModel.getDefaultBaseline())){
+            if(model.getDefaultBaseline()!=null && BooleanUtils.isTrue(model.getDefaultBaseline()) ){
```

Comment:
i think my intention here is to say if the current baseline is set to default, but in previousMode, it was not default, then go ahead to set it to default. 
this phases out the possibility that it's already a default in previoudModel, 

what's the flaw in this logic that you changed it to only check the current model?

Reply:
The previous & current state mode will trigged during the CFN test of contract_create_update event. In this event wit will first start the create action and it will pass the state to Update handler the update will have details of current state for updated & previous state without updating cause.

Reply:
that doesn't explain why you don't check the previous state, which could essentially reduce a API call if it's already a default baseline before updating handler.

Reply:
Shall i revert back to the original formate.

Reply:
what do you mean, as of now, it still doesn't check the previous state
and wouldn't BooleanUtils already check null, I though so, but i could be wrong

Reply:
i revert back to the original file so it should be good for checking both current and the previous one.